src/main/java/com/example/BillingSystem/service/impl/ProductServiceImpl.java

```diff
@@ -54,6 +54,7 @@ public boolean isProductTypeExists(String productType){
         }
     }
 
+    @Override
     public Product getProductDetails(String productType){
         Product productDetails = new Product();
```

Comment:
Add a null or empty check to the productType to make sure that the input is a valid one. This sort of check to the arguments of the methods can be added in other places too. 
Eg: 
if (productType == null || productType.trim().isEmpty() ) {
   throw new IllegalArgumentException("product type can not be null or empty.");
}